internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -0,0 +1,114 @@
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package delegatingresolver implements the default resolver that creates child
+// resolvers to resolver targetURI as well as proxy address.
+package delegatingresolver
+
+import (
+	"fmt"
+	"os"
+
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/grpc/serviceconfig"
+)
+
+type delegatingResolver struct {
+	target         resolver.Target
+	cc             resolver.ClientConn
+	isProxy        bool
+	targetResolver resolver.Resolver
+	targetAddrs    []resolver.Address
+}
+
+type innerClientConn struct {
+	cc           resolver.ClientConn
```

Comment:
This field could be embedded. Doing so would save you from implementing methods that simply call the same method on the parent.

Reply:
Actually, I don't see the `cc` field directly used anywhere. It is always `parent.cc` that is being used. If `parent.cc` is the only thing that is required, you could still embed that, and save yourself from implementing methods that are simply passed along to the parent.